lib/solargraph/shell.rb

```diff
@@ -232,13 +232,48 @@ def bundle
       puts 'The `bundle` command is deprecated. Solargraph currently uses RBS instead.'
     end
 
+    desc 'cache', 'Cache a gem', hide: true
+    def cache gem, version = nil
+      spec = Gem::Specification.find_by_name(gem, version)
+      pins = GemPins.build(spec)
+      Cache.save('gems', "#{spec.name}-#{spec.version}.ser", pins)
+    end
+
+    desc 'gems', 'Cache documentation for installed gems'
+    option :rebuild, type: :boolean, desc: 'Rebuild existing documentation', default: false
+    def gems *names
+      if names.empty?
+        Gem::Specification.to_a.each do |spec|
+          next unless options.rebuild || !Yardoc.cached?(spec)
+
+          puts "Processing gem: #{spec.name} #{spec.version}"
+          pins = GemPins.build(spec)
+          Cache.save('gems', "#{spec.name}-#{spec.version}.ser", pins)
+        end
+      else
+        names.each do |name|
+          spec = Gem::Specification.find_by_name(name)
+          if spec
+            next unless options.rebuild || !Yardoc.cached?(spec)
+
+            puts "Processing gem: #{spec.name} #{spec.version}"
+            pins = GemPins.build(spec)
+            Cache.save("#{spec.name}-#{spec.version}.ser", pins)
```

Comment:
Is this intended to be a different path than the save() line above?

Reply:
Good catch. I fixed it and pushed a commit.

I think this indicates an issue I should resolve with caching in general. There are now four different types of cached files:

* Three closely related but subtly different caches for core, stdlib, and gem pins
* Yardocs built with yard-solargraph

I'll try to codify the standard for each cache to avoid confusion.